src/Core/Impl/IO/PathUtils.cs

```diff
@@ -45,7 +45,14 @@ public static bool IsValidFileNameCharacter(char character)
         /// Returns true if the path has a directory separator character at the end.
         /// </summary>
         public static bool HasEndSeparator(string path)
-            => !string.IsNullOrEmpty(path) && DirectorySeparators.Contains(path[path.Length - 1]);
+            => !string.IsNullOrEmpty(path) && IsDirectorySeparator(path[path.Length - 1]);
+
+
+        public static bool IsDirectorySeparator(char c) => Array.IndexOf(DirectorySeparators, c) != -1;
+
+        public static bool PathStartsWith(string s, string prefix)
+            => s != null && s.StartsWith(prefix, StringExtensions.PathsStringComparison) &&
+               (s.Length == prefix.Length || IsDirectorySeparator(s[prefix.Length]));
```

Comment:
This is fix for #1321, to ensure that part of folder name isn't considered as part of the path.